src/super_gradients/training/models/segmentation_models/segformer.py

```diff
@@ -0,0 +1,516 @@
+
+        out = self.linear_fuse(torch.cat(out_lst[::-1], dim=1))
+        out = self.linear_pred(self.dropout(out))
+
+        return out
+
+
+# TODO: add support for aux heads? (not in original impl) (currently not using)
+class SegFormer(SegmentationModule):
+    def __init__(
+        self,
+        num_classes: int,
+        encoder_embed_dims: list,
+        encoder_layers: list,
+        eff_self_att_reduction_ratio: list,
+        eff_self_att_heads: list,
+        decoder_embed_dim: int,
+        overlap_patch_size: list,
+        overlap_patch_stride: list,
+        overlap_patch_pad: list,
+        in_channels: int = 3,
+    ):
+        """
+        :param num_classes: number of classes
+        :param encoder_embed_dims: the patch embedding dimensions (number of output channels in each encoder stage)
+        :param encoder_layers: the number of encoder layers in each encoder stage
+        :param eff_self_att_reduction_ratio: the reduction ratios of the efficient self-attention in each stage
+        :param eff_self_att_heads: number of efficient self-attention heads in each stage
+        :param overlap_patch_size:  the patch size of the overlapping patch embedding in each stage
+        :param overlap_patch_stride:  the patch stride of the overlapping patch embedding in each stage
+        :param overlap_patch_pad:  the patch padding of the overlapping patch embedding in each stage
+        :param in_channels:  number of input channels
+        """
+
+        super().__init__(use_aux_heads=False)
+
+        self.encoder_embed_dims = encoder_embed_dims
+
+        self._backbone = MiTBackBone(
+            embed_dims=encoder_embed_dims,
+            encoder_layers=encoder_layers,
+            eff_self_att_reduction_ratio=eff_self_att_reduction_ratio,
+            eff_self_att_heads=eff_self_att_heads,
+            overlap_patch_size=overlap_patch_size,
+            overlap_patch_stride=overlap_patch_stride,
+            overlap_patch_pad=overlap_patch_pad,
+            in_channels=in_channels,
+        )
+
+        self.decode_head = SegFormerHead(encoder_dims=encoder_embed_dims, embed_dim=decoder_embed_dim, num_classes=num_classes)
+
+        self.init_params()
+
+    def init_params(self):
+
+        for m in self.modules():
+            if isinstance(m, nn.Linear):
+                torch.nn.init.trunc_normal_(m.weight, std=0.02)
+                if m.bias is not None:
+                    nn.init.zeros_(m.bias)
+            elif isinstance(m, nn.Conv2d):
+                nn.init.kaiming_normal_(m.weight, mode="fan_out", nonlinearity="relu")
+                if m.bias is not None:
+                    nn.init.constant_(m.bias, 0)
+            elif isinstance(m, (nn.LayerNorm, nn.BatchNorm2d, nn.SyncBatchNorm)):
+                nn.init.ones_(m.weight)
+                nn.init.zeros_(m.bias)
+
+    @property
+    def backbone(self):
+        return self._backbone
+
+    def _remove_auxiliary_heads(self):
```

Comment:
why do we need this?

Reply:
it is an unimplemented abstract method, and it's simply 'pass' since currently aux heads aren't implemented for segformer

Reply:
@ofrimasad also, please have a look at the failed test, not sure what's the cause